service/src/main/java/bio/terra/externalcreds/services/ProviderClientCache.java

```diff
@@ -41,12 +45,32 @@ public void resetCache() {
     log.info("ProviderClientCache reset");
   }
 
-  private ClientRegistration buildClientRegistration(ProviderProperties providerInfo) {
-    var builder =
-        ClientRegistrations.fromOidcIssuerLocation(providerInfo.getIssuer())
-            .clientId(providerInfo.getClientId())
-            .clientSecret(providerInfo.getClientSecret())
-            .issuerUri(providerInfo.getIssuer());
+  private ClientRegistration buildClientRegistration(
+      String providerName, ProviderProperties providerInfo) {
+    ClientRegistration.Builder builder;
+    if (providerName.equals(Provider.RAS.toString())) {
+      builder =
+          ClientRegistrations.fromOidcIssuerLocation(providerInfo.getIssuer())
+              .clientId(providerInfo.getClientId())
+              .clientSecret(providerInfo.getClientSecret())
+              .issuerUri(providerInfo.getIssuer());
+    } else if (providerName.equals(Provider.GITHUB.toString())) {
+      var redirectUri =
```

Comment:
Can a provider have multiple valid redirect uris? If so, I can pass in the redirect uri from the API request instead of getting it from the config.

Reply:
You can configure multiple "Callback URL"'s in the github app but not sure why we would need multiple, I would think getting it from either source would be fine. Maybe getting it from the request makes sense just so if we change it we have to update it in less places?